### ISSUE TITLE
Time object returned by File#mtime doesn't honor TZ environment variable on Windows

### ISSUE THREAD

**Issue by mojavelinux** [jruby/jruby#6659]
**Environment Information**
  
* JRuby version: 9.2.17.0
* JRUBY_OPTS: none
* Operating system and platform: Windows
* Installed gems: n/a
* Environment variables: TZ=UTC

**Expected Behavior**

To match CRuby, File#mtime (and related methods) should honor the TZ environment variable, just like Time.now. This works correctly on Linux. The behavior differs on Windows. Time.now honors the TZ environment variable, whereas File#mtime does not.

Here's an example script to demonstrate the issue:

```rb
puts File.open('test.rb') {|f| f.mtime }.utc_offset
puts Time.now.utc_offset
puts Time.at(0).utc_offset
```

**Actual Behavior**

If you run the script without setting the TZ environment variable, you will get your local utc offset (assuming, of course, your system time is not already set to UTC).

```
set TZ=
jruby test.rb
#=> -21600
#=> -21600
#=> -25200
```

If you run the script with `TZ=UTC`, you should see 0 for all three values. However, on JRuby for Windows, the first value is not 0.

```
set TZ=UTC
jruby test.rb
#=> -21600
#=> 0
#=> 0
```

Here's the result you get on JRuby for Linux, consistent with CRuby.

```
TZ=UTC jruby test.rb
#=> 0
#=> 0
#=> 0
```

**Comment by mojavelinux** [comment]
There is a workaround. You can retrieve the timestamp from the Time object and run it through Time.at:

```rb
Time.at(File.open('test.rb') {|f| f.mtime }.to_i).utc_offset
```

**Comment by ladybando** [comment]
I am a beginner and would like to work on this. To be clear, are you looking to implement the workaround or find a solution to why it is not returning what is expected on a Windows machine?

**Comment by headius** [comment]
Fixed in #6720.

**Comment by headius** [comment]
@ladybando Sorry we never saw your comment! I encourage you to look at other issues or stop by our chat (http://jruby.org/chat) if you are still interested in helping out.
